### PR TITLE
Pr 8341 and payload len fix/v2

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) tickets:
https://redmine.openinfosecfoundation.org/issues/5379
https://redmine.openinfosecfoundation.org/issues/5693

Previous PR: https://github.com/OISF/suricata/pull/8342

Changes since v1:
- add new event type udp.len_invalid

suricata-verify-pr: 1055

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .app_layer.error.tls.parser | 931 | 1145 | 122.99% |
| | SURI_TLPR1_stats_chk | | |
| .app_layer.flow.http | 3118951 | 3340799 | 107.11% |
| .app_layer.flow.tls | 510462 | 623355 | 122.12% |
| .app_layer.flow.ssh | 8937 | 11032 | 123.44% |
| .app_layer.flow.telnet | 1624 | 2334 | 143.72% |
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 207683 | 0.04 | 

Pipeline 11673

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .app_layer.error.tls.parser | 931 | 1145 | 122.99% |
| | SURI_TLPR1_stats_chk | | |
| .app_layer.flow.http | 3118951 | 3340798 | 107.11% |
| .app_layer.flow.tls | 510462 | 623355 | 122.12% |
| .app_layer.flow.ssh | 8937 | 11032 | 123.44% |
| .app_layer.flow.telnet | 1624 | 2334 | 143.72% |
| .app_layer.error.ftp-data.gap | 0 | 1 | - |
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 3859 | 0.00 | 

Pipeline 11677

File: rules/decoder-events.rules [hunk 1]
```diff
@@ -68,6 +68,7 @@ alert pkthdr any any -> any any (msg:"SURICATA TCP duplicated option"; decode-ev
 alert pkthdr any any -> any any (msg:"SURICATA UDP packet too small"; decode-event:udp.pkt_too_small; classtype:protocol-command-decode; sid:2200038; rev:2;)
 alert pkthdr any any -> any any (msg:"SURICATA UDP header length too small"; decode-event:udp.hlen_too_small; classtype:protocol-command-decode; sid:2200039; rev:2;)
 alert pkthdr any any -> any any (msg:"SURICATA UDP invalid header length"; decode-event:udp.hlen_invalid; classtype:protocol-command-decode; sid:2200040; rev:2;)
+alert pkthdr any any -> any any (msg:"SURICATA UDP invalid packet length"; decode-event:udp.len_invalid; classtype:protocol-command-decode; sid:2200220; rev:2;)
```
Comment (victorjulien) [inline rules/decoder-events.rules:71]: invalid length field in the header is the real issue here?

File: src/detect-engine-event.c [hunk 2]
```diff
@@ -160,6 +160,9 @@ static DetectEngineEventData *DetectEngineEventParse (const char *rawstr)
 
     if (de->event == STREAM_REASSEMBLY_OVERLAP_DIFFERENT_DATA) {
         StreamTcpReassembleConfigEnableOverlapCheck();
+    } else if (de->event == UDP_HLEN_INVALID) {
```
Comment (victorjulien) [inline src/detect-engine-event.c:163]: I think this should be implemented using the strict concept as done here: https://github.com/OISF/suricata/blob/master/src/detect-app-layer-event.c#L201

cc @lukashino

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: src/detect-engine-event.c (hunk 2 above)
Comment (victorjulien) [reply src/detect-engine-event.c:163]: @jufajardini this strict concept should make it to the dev guide :)

**Comment by victorjulien** [comment]
replaced by #8404

File: src/detect-engine-event.c (hunk 2 above)
Comment (lukashino) [reply src/detect-engine-event.c:163]: followed the concept and created `OutdatedEvent` function in decode part of the engine.

**Pull request closed without merging**